lib/Onyx.js

```diff
@@ -982,28 +934,20 @@ function applyMerge(key, data) {
  * @returns {Promise}
  */
 function merge(key, value) {
-    if (mergeQueue[key]) {
-        mergeQueue[key].push(value);
-        return Promise.resolve();
-    }
-
-    mergeQueue[key] = [value];
-    return get(key)
-        .then((data) => {
-            try {
-                const modifiedData = applyMerge(key, data);
-
-                // Clean up the write queue so we
-                // don't apply these changes again
-                delete mergeQueue[key];
+    // Logging properties only since values could be sensitive things we don't want to log
+    Logger.logInfo(`emrge() called for key: ${key}${_.isObject(value) ? ` properties: ${_.keys(value).join(',')}` : ''}`);
 
-                return set(key, modifiedData);
-            } catch (error) {
-                Logger.logAlert(`An error occurred while applying merge for key: ${key}, Error: ${error}`);
-            }
+    // Array values are always set never merged because the MergePatch algorithm in SQLite will always replace arrays.
+    // Old concatenating of array values behavior is deprecated.
+    if (_.isArray(value)) {
+        return set(key, value);
+    }
 
-            return Promise.resolve();
-        });
+    return new Promise(resolve => get(key)
+        .then(currentValue => Storage.mergeItem(key, value, currentValue)
+            .then(updatedValue => resolve(
+                notifyAfterWrite(key, updatedValue, Storage.mergeItem),
```

Comment:
`mergeItem` does not resolve  with `updatedValue` it resolves with `Promise.resolve()` - nothing